MapboxNavigation/NavigationMapView.swift

```diff
@@ -473,7 +473,8 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         let allRoutesShape = navigationMapViewDelegate?.navigationMapView(self, shapeFor: routes) ?? shape(for: routes, legIndex: legIndex)
         let allRoutesSource = addAllRoutesSource(style, shape: allRoutesShape)
         
-        let mainRouteCasingShape = navigationMapViewDelegate?.navigationMapView(self, shapeFor: [mainRoute]) ?? shape(for: [mainRoute], legIndex: legIndex)
+        let mainRouteCasingShape = navigationMapViewDelegate?.navigationMapView(self, simplifiedShapeFor: mainRoute) ?? shape(forCasingOf: mainRoute, legIndex: legIndex)
```

Comment:
Should `allRoutesShape` also call `navigationMapView(_:simplifiedShapeFor:)`? What should call `navigationMapView(_:shapeFor:)`? Previously, a developer would’ve implemented `navigationMapView(_:shapeFor:)` to provide the shape data for the whole route (normally to be used as the casing) and `navigationMapView(_:simplifiedShapeFor:)` to provide the shape data for the route broken up by arbitrary segments (normally to be colored by traffic congestion).

Reply:
As I found in the code, `navigationMapView(_:shapeFor:)` is meant to provide route shapes, and `navigationMapView(_:simplifiedShapeFor:)` meant to provide the casing shape. At least when I provided my custom Shapes like that they were colored as route and casing respectfully. Even if my custom shape for `navigationMapView(_:shapeFor:)` had just one `Polyline` it still was colored with congestion segments. 
Doesn't it supposed to be like that?

Reply:
> As I found in the code, `navigationMapView(_:shapeFor:)` is meant to provide route shapes, and `navigationMapView(_:simplifiedShapeFor:)` meant to provide the casing shape.

Yes, sorry, I got the two methods mixed up in https://github.com/mapbox/mapbox-navigation-ios/pull/2623#discussion_r488007899. If you provide only one shape in `navigationMapView(_:shapeFor:)`, I don’t think there should be colored congestion segments.

Reply:
So... Is this convo resolved? :)